### ISSUE TITLE
Expression ordering inconsistent: f(k+j) but (j+k)^2

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#2463]
Imported from SourceForge on 2024-07-06 09:23:10
Created by **[macrakis](https://sourceforge.net/u/macrakis/)** on 2017-11-22 21:36:03
Original: https://sourceforge.net/p/maxima/bugs/3360

---

The canonical display order for j+k is k+j.

Thus, `j+k` => `k+j`, `f(j+k)` => `f(k+j)`, `2*(j+k)` => `2*(k+j)`, `1/(j+k)` => `1/(k+j)`

But the canonical order of `(k+j)^2` is `(j+k)^2` (!!!)

Why is the expression ordering inconsistent?

This means that you can't reliably use "part" when looking at an expression: `part(f(k+j),1,1)`=>`k`, but `part((k+j)^2,1,1)` => `j` !

Is this old or new behavior?

i686-pc-mingw32",lisp_name="SBCL",lisp_version="1.2.7"

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-06 09:23:11
Created by **[macrakis](https://sourceforge.net/u/macrakis/)** on 2017-11-22 22:17:23
Original: https://sourceforge.net/p/maxima/bugs/3360/#6a2e

---

Note: Maxima generally tries hard to keep display and "part" consistent, for example:

powerdisp:false$ part(x^2+x^3,1) => x^3
powerdisp:true$ part(x^2+x^3,1) => x^2

Another fun example of the problem:

(j+k+1)/(j+k+2) => (1+j+k)/(k+j+2)
 1/% => (2+j+k)/(k+j+1)
 
 That is, the ordering depends on whether it's in the numerator or in the denominator.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-06 09:23:15
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2017-11-23 05:15:44
Original: https://sourceforge.net/p/maxima/bugs/3360/#e532

---

- Description has changed:

Diff:

~~~~

--- old
+++ new
@@ -1,12 +1,12 @@
 The canonical display order for j+k is k+j.
 
-Thus, j+k => k+j, f(j+k) => f(k+j), 2*(j+k) => 2*(k+j), 1/(j+k) => 1/(k+j)
+Thus, `j+k` => `k+j`, `f(j+k)` => `f(k+j)`, `2*(j+k)` => `2*(k+j)`, `1/(j+k)` => `1/(k+j)`
 
-But the canonical order of (k+j)^2 is (j+k)^2 (!!!)
+But the canonical order of `(k+j)^2` is `(j+k)^2` (!!!)
 
 Why is the expression ordering inconsistent?
 
-This means that you can't reliably use "part" when looking at an expression: part(f(k+j),1,1)=>k, but part((k+j)^2,1,1) => j !
+This means that you can't reliably use "part" when looking at an expression: `part(f(k+j),1,1)`=>`k`, but `part((k+j)^2,1,1)` => `j` !
 
 Is this old or new behavior?
 

~~~~

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-06 09:23:19
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2017-11-23 05:15:45
Original: https://sourceforge.net/p/maxima/bugs/3360/#e43a

---

I can't reproduce this behavior. I have tried it with 5.10.0 + Clisp 2.49, and current post-5.40 + SBCL 1.3.13 and Clisp 2.49. What version are you working with? 

The most recent commit that might or might not be relevant is this:

~~~~
commit af38d28ca56623451bb94db9afb47eb3c405f82a
Author: Dan Gildea <dgildea>
Date:   Mon Jan 18 14:53:52 2016 -0500

    make orderlessp transitive
    
    fixes bug [#3072]
    
    src/simp.lisp:
    o ordfna: use ordhack consistently whenever we see mplus/mtimes
    
    tests/testsuite.lisp
    o test added for bug [#3072] all pass now
~~~~

I didn't look beyond that.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-06 09:23:22
Created by **[macrakis](https://sourceforge.net/u/macrakis/)** on 2017-11-25 18:04:25
Original: https://sourceforge.net/p/maxima/bugs/3360/#6db1

---

Sorry about the incomplete version. It's version="5.37.2",
timestamp="2015-09-18 21:03:17",host="i686-pc-mingw32",lisp_name="SBCL",lisp_version="1.2.7".

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-06 09:23:26
Created by **[macrakis](https://sourceforge.net/u/macrakis/)** on 2017-11-25 20:11:02
Original: https://sourceforge.net/p/maxima/bugs/3360/#0a65

---

Fixed in (version="5.38.1_5_gdf93b7b_dirty",timestamp="2016-05-13 22:33:49",host=
"i686-w64-mingw32",lisp_name="CLISP",lisp_version=
"2.49 (2010-07-07) (built on toshiba [192.168.43.3])")

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-06 09:23:29
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2017-11-26 01:44:28
Original: https://sourceforge.net/p/maxima/bugs/3360/#c0e3

---

- **status**: open --> closed
